src/Magnum/Math/Geometry/Test/IntersectionBenchmark.cpp

```diff
@@ -0,0 +1,156 @@
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include <Corrade/TestSuite/Tester.h>
+
+#include "Magnum/Math/Geometry/Intersection.h"
+#include "Magnum/Math/Angle.h"
+#include "Magnum/Math/Matrix4.h"
+
+#include <random>
+#include <tuple>
+#include <utility>
+
+namespace Magnum { namespace Math { namespace Geometry { namespace Benchmark {
+
+/* @brief Ground truth, slow sphere cone intersection - calculating exact distances,
+ *        no optimizations, no precomputations
+ * @param sphereCenter Sphere center
+ * @param radius Sphere radius
+ * @param origin Origin of the cone
+ * @param normal Cone normal
+ * @param angle Cone opening angle (0 < angle < pi).
+ *
+ * Returns `true` if the sphere intersects with the cone. */
+template<class T> bool sphereConeGT(
+        const Math::Vector3<T>& sphereCenter, const T radius,
+        const Math::Vector3<T>& origin, const Math::Vector3<T>& normal, const Math::Rad<T> angle) {
+    const Math::Vector3<T> diff = sphereCenter - origin;
+    const Math::Vector3<T> dir = diff.normalized();
+    const Math::Rad<T> halfAngle = angle/T(2);
+
+    /* Compute angle between normal and point */
+    const Math::Rad<T> actual = Math::acos(dot(normal, dir));
+
+    /* Distance from cone surface */
+    const T distanceFromCone = Math::sin(actual - halfAngle)*diff.length();
+
+    /* Either the sphere center lies in cone, or cone is max radius away from the cone */
+    return actual <= halfAngle || distanceFromCone <= radius;
+}
+
+template<class T>
+Math::Matrix4<T> coneViewFromCone(const Math::Vector3<T>& origin, const Math::Vector3<T>& normal) {
+    return Math::Matrix4<T>::lookAt(origin, origin + normal, Math::Vector3<T>::yAxis()).inverted();
+}
+
+typedef Math::Vector2<Float> Vector2;
+typedef Math::Vector3<Float> Vector3;
+typedef Math::Vector4<Float> Vector4;
+typedef Math::Matrix4<Float> Matrix4;
+typedef Math::Frustum<Float> Frustum;
+typedef Math::Deg<Float> Deg;
+typedef Math::Rad<Float> Rad;
+
+struct IntersectionBenchmark: Corrade::TestSuite::Tester {
+    explicit IntersectionBenchmark();
+
+    void sphereConeNaive();
+    void sphereCone();
+    void sphereConeView();
+
+    std::vector<std::pair<Vector3, Float>> _spheres;
+    std::vector<std::tuple<Vector3, Vector3, Rad>> _cones;
+};
+
+IntersectionBenchmark::IntersectionBenchmark() {
+    addBenchmarks({&IntersectionBenchmark::sphereConeNaive,
+                   &IntersectionBenchmark::sphereCone,
+                   &IntersectionBenchmark::sphereConeView}, 1000);
```

Comment:
Wouldn't `100` be enough? I.e., don't make this run more than ~5 seconds on the CI and ~1 second on your machine. It's always possible to increase the iteration count using [--repeat-every](http://doc.magnum.graphics/corrade/classCorrade_1_1TestSuite_1_1Tester.html#TestSuite-Tester-command-line), if you want less variance in measured times.

Also, I think decreasing this number but increasing the loop count inside `CORRADE_BENCHMARK(n)` would help with the variance too -- so, for example 25 here and 50 there. You have to experiment, though :)